overlord/devicestate/devicemgr.go

```diff
@@ -395,16 +401,21 @@ func (m *DeviceManager) ensureUbuntuSaveSnapFolders() error {
 // setupUbuntuSave sets up ubuntu-save partition. It makes sure
 // to mount ubuntu-save (if feasible), and ensures the correct snap
 // folders are present according to currently installed snaps.
-func (m *DeviceManager) setupUbuntuSave() error {
+func (m *DeviceManager) setupUbuntuSave(dev snap.Device) error {
 	if err := m.ensureUbuntuSaveIsMounted(); err != nil {
 		return err
 	}
 
 	// At this point ubuntu-save should be available under the
 	// /var/lib/snapd/save path, so we mark the partition as such.
+	m.saveAvailable = true
+
 	// The last step is to ensure needed folder structure is present
 	// for the per-snap folder storage.
-	m.saveAvailable = true
+	// We support this only on Core for now.
```

Comment:
Are we sure we will never user install mode on classic? Maybe better to be on the safe side here?

Reply:
the questions is really whether we want to give save folders to snaps on classic. The answer for now is no, I could think of giving it maybe to the gadget but we don't run the install-device on classic either atm. OTOH this remind me that I need to a make change in snapstate as well in a follow up